catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get/cip509.rs

```diff
@@ -0,0 +1,86 @@
+//! Cardano Improvement Proposal 509 (CIP-509) API object.
+//! Doc Reference: <https://github.com/input-output-hk/catalyst-CIPs/tree/x509-envelope-metadata/CIP-XXXX>
+//! CDDL Reference: <https://github.com/input-output-hk/catalyst-CIPs/blob/x509-envelope-metadata/CIP-XXXX/x509-envelope.cddl>
+
+use poem_openapi::{
+    types::{Example, ToJSON},
+    Object,
+};
+
+use crate::service::common::{
+    objects::generic::json_object::JSONObject,
+    types::{
+        array_types::impl_array_types,
+        cardano::{slot_tx_idx::SlotTxnIdx, transaction_id::TxnId},
+        generic::uuidv4::UUIDv4,
+    },
+};
+
+/// CIP 509 registration transaction data.
+#[derive(Debug, Clone, Object)]
+#[oai(example = true)]
+pub(crate) struct Cip509 {
```

Comment:
Curious, besides from the data from the db, do we need to return other registration information?

Reply:
It will be updated and extended later, for now its just something to start implementing the endpoint itself